### ISSUE TITLE
[ENH] neural network libraries in thuml time-series-library

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#7243]
I found this repository

https://github.com/thuml/Time-Series-Library

which implements a large number of state-of-art neural network models in `torch`, and seems to be updated regularly.

The repository is not a pypi package and relies on source install.

There is also a large PR that turns it into a package, but it seems stale or unreviewed:
https://github.com/thuml/Time-Series-Library/pull/457

I wonder about the best way to integrate here. Nothing keeps us from copying layers and networks, but that is perhaps not the best way to proceed (and also not that nice towards the authors). But interfacing is not possible without proper package management, similar to the `cure-lab` problem.

From a tech perspective, a merge or some kind of connection with `pytorch-forecasting` could make sense, and then interfacing from `sktime` one layer higher. A connection which is not merge is turning `pytorch-forecasting` into a `torch` neural network and components indexer, using `scikit-base`, where components are not necessarily directly in the package.

FYI @XinyuWuu, the most active devs seem to be all from China, do you know them? I have also not reviewed the code in detail, e.g., intersections with `pytorch-forecasting`, or with `neuralforecast`.

**Comment by fkiraly** [comment]
FYI @thawn, would you like to help with integration with `pytorch-forecasting`?

* we are now maintaining `pytorch-forecasting`
* your pull request for proper package refactor looks like the direction we should go!

**Comment by thawn** [comment]
@fkiraly I'd be happy to help. How would you like to proceed?

Are you by any chance at the neurips 20204 in Vancouver? Then we could meet up ;-)

**Comment by fkiraly** [comment]
> Are you by any chance at the Neurips 2024 in Vancouver?

No, not there. I am in the middle of a house move. Maybe someone else of @sktime/core-developers is?

> How would you like to proceed?

Concretely, I would suggest getting in touch via discord (sktime server) https://discord.com/invite/54ACzaFsn7,
there is a channel for `pytorch-forecasting` there.

I think we need to have a 30-60min meeting with other deep learning focused developers to get to a clear architectural target state for `pytorch-forecasting` and a potential integration.

If you already have concrete ideas, that would be great to hear.

@agobbifbk will likely be driving the rework of ptf, he will also be at our meetup on Dec 20 (13 UTC). We could meet before quickly, and then with Andrea on Dec 20?

**Comment by fkiraly** [comment]
FYI @wuhaixu2016, @ZDandsomSP, @GuoKaku, @DigitalLifeYZQiu, @Musongwhk - discussions have evolved towards integrating the `torch` layers first, see https://github.com/sktime/pytorch-forecasting/issues/1736
